### PR TITLE
Glossary initial commit

### PR DESCRIPTION
**Pull request by ZKRobi**

### CODE REVIEW

**Review by BalassaMarton — APPROVED** [review]

**Review by ztanczos — APPROVED** [review]

**Review by psmulovics — COMMENTED** [review]
Do we want to move glossary under docs/ or architecture/?

**Comment by kruplm** [comment]
> Do we want to move glossary under docs/ or architecture/?

No need. Most OSS projects have the glossary.md (or a folder) in the root, so it's easy to find.

**Comment by ztanczos** [comment]
> Do we want to move glossary under docs/ or architecture/?

I wouldn't move it under architecture. If / once we'd have significant amount of docs then we can re-visit whether to put glossary there.

File: Glossary/Plugin.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Plugin
```
Comment (BalassaMarton) [inline Glossary/Plugin.md:1]: Spaces are missing after `#` (it is needed for correct rendering)

**Comment by BalassaMarton** [comment]
Minor thing: can we use capitalization consistently? Most of the root directories are all lower case, with some irritating exceptions.

**Comment by ZKRobi** [comment]
> 

Sounds good, but which one to use? :)

**Review by kruplm — APPROVED** [review]
LGTM.

File: Glossary/Plugin.md (hunk 1 above)
Comment (ZKRobi) [reply Glossary/Plugin.md]: fixed

**Review by BalassaMarton — APPROVED** [review]

File: glossary/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Glossary
+This folder contains terms used in the Compose project to describe certain concepts.
+
+[Application](Application.md)\
```
Comment (BalassaMarton) [inline glossary/README.md:4]: I say we drop the links because we will forget to update the list for sure. The link to Manifest is missing already.

Comment (ztanczos) [reply glossary/README.md:4]: Can this be auto-generated somehow?

Comment (ZKRobi) [reply glossary/README.md:4]: I'm dropping the links for now, and I'll look into a solution for generating them. I could only find a solution for generating a table of contents within a single file when I did a quick search

**Review by kruplm — APPROVED** [review]

**Pull request merged**